OpenKh.Tests/Patcher/PatcherTests.cs

```diff
@@ -514,6 +514,638 @@ public void MergeKh2AreaDataScriptTest()
+                        Ability = 127
+                        }
+                    }
+                });
+                writer.Flush();
+            });
+
+            patcher.Patch(AssetsInputDir, ModOutputDir, patch, ModInputDir);
+
+            AssertFileExists(ModOutputDir, "00battle.bar");
+
+            File.OpenRead(Path.Combine(ModOutputDir, "00battle.bar")).Using(stream =>
+            {
+                var binarc = Bar.Read(stream);
+                var fmlvStream = Kh2.Battle.Fmlv.Read(binarc[0].Stream);
+                Assert.True(fmlvStream[0].Exp == 5);
+                Assert.True(fmlvStream[0].Ability == 127);
+            });
+
+        }
+
+        [Fact]
+        public void ListReplaceBonsTest()
+        {
+            var patcher = new PatcherProcessor();
+            var serializer = new Serializer();
+            var patch = new Metadata()
+            {
+                Assets = new List<AssetFile>()
+                {
+                    new AssetFile()
+                    {
+                        Name = "00battle.bar",
+                        Method = "binarc",
+                        Source = new List<AssetFile>()
+                        {
+                            new AssetFile()
+                            {
+                                Name = "bons",
+                                Method = "listreplace",
+                                Type = "List",
+                                Source = new List<AssetFile>()
+                                {
+                                    new AssetFile()
+                                    {
+                                        Name = "BonsList.yml",
+                                        Type = "bons"
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+
+            File.Create(Path.Combine(AssetsInputDir, "00battle.bar")).Using(stream =>
+            {
+                var bonsEntry = new List<Kh2.Battle.Bons>()
+                {
+                    new Kh2.Battle.Bons
+                    {
+                        CharacterId = 1,
+                        RewardId = 15,
+                        BonusItem1 = 10
+                    }
+                    };
+                using var bonsStream = new MemoryStream();
+                Kh2.Battle.Bons.Write(bonsStream, bonsEntry);
+                Bar.Write(stream, new Bar() {
+                    new Bar.Entry()
+                    {
+                        Name = "bons",
+                        Type = Bar.EntryType.List,
+                        Stream = bonsStream
+                    }
+                });
+            });
+
+            File.Create(Path.Combine(ModInputDir, "BonsList.yml")).Using(stream =>
+            {
+                var writer = new StreamWriter(stream);
+                writer.WriteLine("15Sora:");
```

Comment:
It is not 100% clear to me why there is a `15` before the name of the character? 🤔

Reply:
This was my way of creating a unique key of `BonusID` and `Character`. However, in keeping with other changes we've made, this should likely change now.

I will change this to 
```
BonusID:
- Character:
  - Reward ID:
    ...
- Character:
```